gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java

```diff
@@ -261,6 +277,9 @@ public String getApiEntrypointsListenerType(GenericApiEntity genericApiEntity) {
         ) {
             return "HTTP";
         }
+        if (genericApiEntity.getDefinitionVersion() == DefinitionVersion.V4 && genericApiEntity instanceof NativeApiEntity) {
+            return "KAFKA";
+        }
```

Comment:
This is necessary ? For me the rule here it's the same for any V4 api we check the first listener type.